### PR TITLE
Setup associations

### PR DESCRIPTION
**Pull request by thecodechaser**
## This branch includes the following features:
-  Created classes Classroom, Book, Rental.
-  Created the has-many/belongs-to relationship between Classroom and Student.
-  Created the many-to-many relationship between Person and Book using the intermediate class Rental.
-  Modified the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (FrederickMih) [inline book.rb]: - [ ] Hi @thecodechaser, please let's add another method of adding rentals. To achieve the `has-many `relationship association, the book itself should be sent as a parameter to create the new rental. You may consider doing this:

```ruby
def add_rental(person, date)
      Rental.new(date, person, self)
end
```

File: person.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 require './corrector'
+require './rental'
 
 class Person
```
Comment (FrederickMih) [inline person.rb:4]: - [ ] A method for adding rentals should be present in the Person class and the person itself should be sent as a parameter to create the new rental achieving the `has-many` association. The piece of code below may be helpful to you.

```ruby
def add_rental(book, date)
      Rental.new(date, book, self)
end
```

**Review by FrederickMih — CHANGES_REQUESTED** [review]
# Hi @thecodechaser , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @thecodechaser, 

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**